lib/lib.rs

```diff
@@ -100,6 +101,26 @@ use types::{Deploy, DeployHash, MAX_SERIALIZED_SIZE_OF_DEPLOY};
 pub use validation::ValidateResponseError;
 pub use verbosity::Verbosity;
 
+/// Puts a [`SpeculativeExec`] to the network for execution.
+///
+/// Sends a JSON-RPC `speculative_exec` request to the specified node.
+///
+/// For details of the parameters, see [the module docs](crate#common-parameters).
+pub async fn speculative_exec(
+    rpc_id: JsonRpcId,
+    node_address: &str,
+    maybe_block_identifier: Option<BlockIdentifier>,
+    verbosity: Verbosity,
+    deploy: Deploy,
+) -> Result<SuccessResponse<PutDeployResult>, Error> {
```

Comment:
This needs to return a `SpeculativeExecResult` rather than a `PutDeployResult`.

As with other result types, it should be exported via `lib/rpcs/results.rs`.

Reply:
Done.